WIP community digest view and email

s-kennedy opened the pull request:
I don't know why the other community digest branch was getting that gpg error on Travis, so i'm copying over the changes to a new branch.

dirkcuys commented on studygroups/models.py:
```diff
@@ -568,3 +576,104 @@ def report_data(start_time, end_time, team=None):
 
+def get_json_response(url):
+    response = requests.get(url)
+    return response.json()
```
I think `response.json()` will throw `json.decoder.JSONDecodeError` if the response body doesn't contain valid json. This will happen with responses like 404

dirkcuys commented on studygroups/templatetags/extras.py:
```diff
@@ -30,3 +31,7 @@ def is_organizer(user):
     return user.is_staff or TeamMembership.objects.filter(user=user, role=TeamMembership.ORGANIZER)
 
+@register.filter
+def category_title(id, categories):
```
I'm not a big fan of template filters, but I guess this is okay. When template filters get used a lot, it easily hides a lot of logic and dependencies. In this case it seems like you're just looking up data in a dictionary that came from the Discourse API, so that's fine.

dirkcuys commented on studygroups/models.py:
```diff
@@ -165,6 +165,7 @@ class StudyGroup(LifeTimeTrackingModel):
     city = models.CharField(max_length=256)
     region = models.CharField(max_length=256, blank=True) # schema.org. Algolia => administrative
     country = models.CharField(max_length=256, blank=True)
+    country_en = models.CharField(max_length=256, blank=True, null=True)
```
@s-kennedy don't set `null=True` on `CharFields`.

See https://docs.djangoproject.com/en/2.1/ref/models/fields/#null

> Avoid using null on string-based fields such as CharField and TextField. If a string-based field has null=True, that means it has two possible values for “no data”: NULL, and the empty string. In most cases, it’s redundant to have two possible values for “no data;” the Django convention is to use the empty string, not NULL.

dirkcuys commented on studygroups/tasks.py:
```diff
@@ -568,28 +569,30 @@ def send_weekly_update():
+    iso_week = today.isocalendar()[1]
+    if iso_week % 3 == 0:
+        end_date = today
+        start_date = end_date - datetime.timedelta(days=14)
```
@s-kennedy shouldn't this be 21 days? 
```python
start_date = end_date - datetime.timedelta(days=21)
```

s-kennedy replied:
yup! thanks :)

s-kennedy commented:
@dirkcuys just a couple of notes:
- I ended up going with adding a `country_en` field becuase I realized that the default country from the Algolia API is using the browser language, so it might not always be the language of the country. 
- I also realized that the option to create a location in the place selector was broken, so I removed it. If we want to let users create a location, we'll need to add fields for city, region, and country. 
- I also added the task to add the English translation to the studygroup, based on your geolocation task.
